### PR TITLE
New serverless pattern - DocumentDB Event Source to Lambda

### PR DESCRIPTION
**Pull request by webster5197**
*Issue #2132 

*Description of changes:*
New SAM Template that creates an AWS Lambda function to process events in an Amazon DocumentDB (with MongoDB compatibility) change stream by configuring an Amazon DocumentDB cluster as an event source 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by webster5197** [comment]
Thank you for the suggestions. I have submitted new commit
Modified the grammatical changes suggested
Added Dynamic Secret Manager values
Edited Inline Code to External
Modified IAM role with dedicated resources

File: docdb-lambda/README.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+1. Setup an EC2 Instance / AWS Cloud9 in the same VPC and configure the Security Group for connecting to the DocumentDB Cluster. 
+    The Default Outbound Rule allows All Traffic to everything, which should be fine for our scenario.
+    [AWS Documentation](https://docs.aws.amazon.com/lambda/latest/dg/with-documentdb-tutorial.html#docdb-cloud9-environment)
+
+2. Install Mongo Shell in your EC2 / Cloud9 by executing below commands in the terminal:
+    
+    Create the MongoDB repository file 
+    ```
+    echo -e "[mongodb-org-4.0] \nname=MongoDB Repository\nbaseurl=https://repo.mongodb.org/yum/amazon/2013.03/mongodb-org/4.0/x86_64/\ngpgcheck=1 \nenabled=1 \ngpgkey=https://www.mongodb.org/static/pgp/server-4.0.asc" | sudo tee /etc/yum.repos.d/mongodb-org-4.0.repo
```
[the remark this answers is not in the document]
Comment (webster5197) [reply docdb-lambda/README.md:77]: Tried to explain. This was picked up from the AWS Tutorial docs

File: docdb-lambda/README.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    * Enter a stack name
+    * Enter the desired AWS Region
+    * Enter a name for your DocumentDB Cluster
+    * Enter a name for your DocumentDB Instance
+    * Enter the DocumentDB Instance Class (Eg. db.t3.medium)
+    * Enter the VPC ID where the resources needs to be created
+    * Enter the First Private Subnet ID that belongs to the above VPC
+    * Enter the Second Private Subnet ID that belongs to the above VPC
+    * Enter the Username for AWS Secrets manager to authenicate with Amazon DocumentDB
+    * Enter the Password for AWS Secrets manager to authenicate with Amazon DocumentDB
```
Comment (bfreiberg) [inline docdb-lambda/README.md:42]:
```suggestion
    * Enter the Username for AWS Secrets manager to authenticate with Amazon DocumentDB
    * Enter the Password for AWS Secrets manager to authenticate with Amazon DocumentDB
```

File: docdb-lambda/README.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    ```
+1. During the prompts:
+    * Enter a stack name
+    * Enter the desired AWS Region
+    * Enter a name for your DocumentDB Cluster
+    * Enter a name for your DocumentDB Instance
+    * Enter the DocumentDB Instance Class (Eg. db.t3.medium)
+    * Enter the VPC ID where the resources needs to be created
+    * Enter the First Private Subnet ID that belongs to the above VPC
+    * Enter the Second Private Subnet ID that belongs to the above VPC
```
Comment (bfreiberg) [inline docdb-lambda/README.md:40]:
```suggestion
    * Enter the first private subnet ID that belongs to the above VPC
    * Enter the second private subnet ID that belongs to the above VPC
```

**Comment by bfreiberg** [comment]
Thanks for the update, there are some minor typos but otherwise looks good.

**Comment by bfreiberg** [comment]
One more thing, please rename the folder to `docdb-lambda-nodejs-cdk`

**Comment by webster5197** [comment]
Thank you for the suggestions. 
Corrected the requested typo changes.
Also changed the folder named to docdb-lambda-nodejs-sam as I am using SAM and not CDK.

**Comment by bfreiberg** [comment]
Thanks for your contribution. A Developer Advocate will merge the changes and publish the pattern to [serverlessland.com](http://serverlessland.com/).

**Pull request merged**